File: 백준/Silver/4396. 지뢰 찾기/지뢰 찾기.c
#include <stdio.h>

int main(void) { 
   int a, result, flag = 0;
   scanf("%d", &a);

   char input_arr[a][a + 1];
   char map[a][a + 1];
   for(int i = 0; i < a; i++){
      scanf("%s", input_arr[i]);
      input_arr[i][a + 1] = '\0';
   }

   for(int i = 0; i < a; i++){
      scanf("%s", map[i]);
      map[i][a + 1] = '\0';
   }

   char arr[a][a + 1];
   for(int i = 0; i < a; i++){
      for(int j = 0; j < a; j++){
         if(map[i][j] == 'x'){
            if(input_arr[i][j] == '*'){
               arr[i][j] = '*';
            }
            else{
               arr[i][j] = '0';
            }
         }
         else{
            arr[i][j] = '.';
         }
      }
      arr[i][a + 1] = '\0';
   }

   for(int i = 0; i < a; i++){
      for(int j = 0; j < a; j++){
         result = 0;
         if(arr[i][j] == '0'){
            if(i > 0 && j > 0 && input_arr[i - 1][j - 1] == '*') result += 1;
            if(i > 0 && input_arr[i - 1][j] == '*') result += 1;
            if(i > 0 && j < a - 1 && input_arr[i - 1][j + 1] == '*') result += 1;
            if(j > 0 && input_arr[i][j - 1] == '*') result += 1;
            if(j < a - 1 && input_arr[i][j + 1] == '*') result += 1;
            if(i < a - 1 && j > 0 && input_arr[i + 1][j - 1] == '*') result += 1;
            if(i < a - 1 && input_arr[i + 1][j] == '*') result += 1;
            if(i < a - 1 && j < a - 1 && input_arr[i + 1][j + 1] == '*') result += 1;

            arr[i][j] = result + '0';
         }
         else if(arr[i][j] == '*'){
            flag = 1;
         }
      }
   }

   for(int i = 0; i < a; i++){
      for(int j = 0; j < a; j++){
         if(flag && input_arr[i][j] == '*'){
            printf("*");
         }
         else{
            printf("%c", arr[i][j]);
         } 
      }
      printf("\n");
   }

   return 0;
}

File: 백준/Silver/4396. 지뢰 찾기/지뢰 찾기.py
a = int(input())

result = [[] for _ in range(a)]
mine_map = []
open_map = []
flag = False
for _ in range(a):
  mine_map.append(list(input()))

for _ in range(a):
  open_map.append(list(input()))

coor = [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1)]

for i in range(a):
  for j in range(a):
    if open_map[i][j] == 'x':
      if mine_map[i][j] == '*':
        flag = True
      
      mine = 0
      for item in coor:
        if i + item[0] >= 0 and j + item[1] >= 0 and i + item[0] <= a - 1 and j + item[1] <= a - 1:
          if mine_map[i + item[0]][j + item[1]] == '*':
            mine += 1
        
      result[i].append(mine)
    
    else:
      result[i].append('.')

for i in range(a):
  for j in range(a):
    if flag and mine_map[i][j] == '*':
      print('*', end = '')
    else:
      print(result[i][j], end = '')
  print()